### PR TITLE
Made RunnableAction call super after execution. Other things

### PR DESCRIPTION
**Pull request by Redrield**
Replaced if tree with switch statement

### CODE REVIEW

**Comment by codeHusky** [comment]
Could this only call super if the runnableaction returns true or something?

**Comment by Redrield** [comment]
Forcing players to add a return at the end of the tunnel would bloat runnable declarations

**Comment by codeHusky** [comment]
I’ll merge and add my changes to be as non invasive as possible

**Comment by FerusGrim** [comment]
This has already been merged, but I found one change you made hilarious. @Redrield 

I actually considered the if/else/case change: http://i.imgur.com/BjBt0gB.png

File: src/main/java/com/codehusky/huskyui/states/action/Action.java [hunk 1]
```diff
@@ -31,10 +32,14 @@
  */
 public class Action {
 
-    @Nonnull private final StateContainer container;
-    @Nonnull private final Player observer;
-    @Nonnull private final ActionType type;
-    @Nonnull private final String goalState;
+    @Nonnull
```
Comment (FerusGrim) [inline src/main/java/com/codehusky/huskyui/states/action/Action.java]: While I appreciate people having differing opinions on formatting, this change makes the rest of the classes I rewrote inconsistent with this single file.

Comment (FerusGrim) [inline src/main/java/com/codehusky/huskyui/states/action/Action.java]: While I appreciate people having differing opinions on formatting, this change makes the rest of the classes I rewrote inconsistent with this single file.

Comment (Redrield) [inline src/main/java/com/codehusky/huskyui/states/action/Action.java]: Sorry, was just auto formatting and the default style guide formats like that. Wasn't intentional

**Pull request merged**